ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -708,6 +708,13 @@ void fixMissingStoreKeys(ConnectedChannel connectedChannel, List<RemoteReplicaIn
       GetResponse getResponse =
           getMessagesForMissingKeys(connectedChannel, exchangeMetadataResponseList, replicasToReplicatePerNode,
               remoteNode, remoteColoGetRequestForStandby);
+      if (remoteColoGetRequestForStandby) {
+        // If we are fetching missing keys for standby replicas (timed out waiting for their keys to arrive from leaders)
+        // during leader-based replication, convert (and cache) received keys in the GET response. This is needed as the
+        // StoreKeyConverter would have cleared these keys from its cache as it is replicating with other replicas before
+        // time out happened for these standby replicas.
+        batchConvertGetResponseKeys(getResponse);
```

Comment:
If we move this to handling no progress replicas code block (line 438 ~470), we might be able to re-use `batchConvertReplicaMetadataResponseKeys` without creating a new method.
It seems the one minor change would suffice: change argument of `batchConvertReplicaMetadataResponseKeys` from `ReplicaMetadataResponse` to `List<ExchangeMetadataResponse>`